packages/astro/src/core/util.ts

```diff
@@ -25,6 +25,10 @@ export function isValidURL(url: string): boolean {
 	return false;
 }
 
+export function isObject(value: unknown): value is Record<string, any> {
+	return Object.prototype.toString.call(value) === '[object Object]';
```

Comment:
This is a bit of a nit, but I think `typeof value === 'object' && value != null` is going to be more accurate because `({ get [Symbol.toStringTag]() { return 'Foo' } }).toString()` won't pass this current check.

Reply:
So the intention here was to check for a basic config object `{...}` because we then tell TS that it has type `Record<string, any>`. Do think that's an okay TS definition for any object, even a Class or `({ get [Symbol.toStringTag]() { return 'Foo' } })`? If so, I'm okay converting to the `typeof value === 'object' && value != null` check you proposed.

Reply:
Update: Converted to `typeof value === 'object' && value != null` since it doesn't really matter for this usecase. If someone wanted to pass some funky object, they should be allowed to :)